drivers/iio/adc/adrv902x/adrv9025.c

```
@@ -1345,13 +1345,63 @@ static const struct iio_info adrv9025_phy_info = {
 	.attrs = &adrv9025_phy_attribute_group,
 };
 
+static ssize_t adrv9025_rx_qec_status_read(adi_adrv9025_Device_t       *device,
+					   adi_adrv9025_RxChannels_e   rxChannel,
+					   char *buf)
+{
+	adi_adrv9025_RxQecStatus_t rxQecStatus = { 0 };
+	int ret, len = 0;
+
+	if (rxChannel < ADI_ADRV9025_ORX1)
+		ret = adi_adrv9025_TrackingCalRxQecStatusGet(device, rxChannel, &rxQecStatus);
+	else
+		ret = adi_adrv9025_TrackingCalOrxQecStatusGet(device, rxChannel, (adi_adrv9025_OrxQecStatus_t *)&rxQecStatus);
+	if (!ret)
+		len = sprintf(buf, "err %d %% %d perf %d iter cnt %d update cnt %d\n", rxQecStatus.errorCode, rxQecStatus.percentComplete,
+			      rxQecStatus.selfcheckIrrDb, rxQecStatus.iterCount, rxQecStatus.updateCount);
+
+	return len;
```

Flip the. You can get rid of `len` and the typical pattern is to first look for errors... You're also ignoring errors...

```
if (ret)
    return ret;

return sprintf(...);
```